File: backend/app/schemas/brief.py
"""Brief + score schemas."""

from __future__ import annotations

from typing import Any

from pydantic import BaseModel

from app.schemas.common import TimestampedModel


class BriefRead(TimestampedModel):
    id: str
    scan_id: str
    account_id: str
    title: str
    executive_summary: str | None = None
    why_now: str | None = None
    key_evidence_json: list[dict[str, Any]] | None = None
    risks_json: list[str] | None = None
    recommended_next_steps_json: list[str] | None = None


class ScoreRead(TimestampedModel):
    id: str
    scan_id: str
    account_id: str
    fit_score: int
    timing_score: int
    relationship_score: int
    evidence_score: int
    total_score: int
    sales_ready: bool
    score_reasoning_json: dict[str, Any] | None = None


class AccountDetail(BaseModel):
    """Enriched account detail returned by GET /accounts/{id}."""

    account: Any  # AccountRead but importing creates a cycle; resolved at use site.
    latest_scan: Any | None = None
    latest_score: ScoreRead | None = None
    latest_brief: BriefRead | None = None
    recent_signals: list[Any] = []

File: frontend/lib/types/brief.ts
// Source: backend/app/schemas/brief.py

import type { Timestamped } from "./common";

export type ScoreRead = Timestamped & {
  id: string;
  scan_id: string;
  account_id: string;
  fit_score: number;
  timing_score: number;
  relationship_score: number;
  evidence_score: number;
  total_score: number;
  sales_ready: boolean;
  score_reasoning_json?: Record<string, unknown> | null;
};

/**
 * Unified, modality-aware account score (web + media).
 * Source of the headline number; `conversation_delta` is the spoken-evidence
 * contribution when `source === "media_scan"`.
 */
export type AccountScoreSnapshot = {
  id: string;
  account_id: string;
  fit_score: number;
  timing_score: number;
  relationship_score: number;
  evidence_score: number;
  total_score: number;
  sales_ready: boolean;
  source: "web_scan" | "media_scan";
  conversation_delta?: number | null;
  reasoning_json?: Record<string, unknown> | null;
  created_at?: string | null;
};

export type BriefEvidenceItem = {
  text: string;
  url?: string;
  source?: string;
  evidence_id?: string;
} & Record<string, unknown>;

export type BriefRead = Timestamped & {
  id: string;
  scan_id: string;
  account_id: string;
  title: string;
  executive_summary?: string | null;
  why_now?: string | null;
  key_evidence_json?: BriefEvidenceItem[] | null;
  risks_json?: string[] | null;
  recommended_next_steps_json?: string[] | null;
};
